server/src/main/java/org/elasticsearch/index/engine/SearchOnlyEngine.java

```diff
@@ -0,0 +1,315 @@
+ * under the License.
+ */
+
+package org.elasticsearch.index.engine;
+
+import org.apache.lucene.index.SegmentInfos;
+import org.apache.lucene.search.SearcherManager;
+import org.apache.lucene.store.AlreadyClosedException;
+import org.elasticsearch.common.lease.Releasable;
+import org.elasticsearch.common.lease.Releasables;
+import org.elasticsearch.common.util.concurrent.ReleasableLock;
+import org.elasticsearch.core.internal.io.IOUtils;
+import org.elasticsearch.index.seqno.SeqNoStats;
+import org.elasticsearch.index.translog.Translog;
+import org.elasticsearch.index.translog.TranslogStats;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.function.BiFunction;
+import java.util.stream.Stream;
+
+/**
+ * A minimal engine that does not accept writes, and always points stats, searcher to the last commit.
+ */
+public final class SearchOnlyEngine extends Engine {
+    private final SegmentInfos lastCommittedSegmentInfos;
+    private final SeqNoStats seqNoStats;
+    private final TranslogStats translogStats;
+    private final SearcherManager searcherManager;
+    private final Searcher lastCommitSearcher;
+
+    public SearchOnlyEngine(Engine engine) throws IOException {
+        super(engine.config());
+        engine.refresh("lockdown");
+        this.seqNoStats = engine.getSeqNoStats(engine.getLastSyncedGlobalCheckpoint());
+        this.translogStats = engine.getTranslogStats();
+        this.lastCommittedSegmentInfos = engine.getLastCommittedSegmentInfos();
+        Searcher searcher = engine.acquireSearcher("lockdown", SearcherScope.INTERNAL);
+        try {
+            this.searcherManager = new SearcherManager(searcher.getDirectoryReader(),
+                new RamAccountingSearcherFactory(engineConfig.getCircuitBreakerService()));
+            this.lastCommitSearcher = searcher;
+            searcher = null;
+        } finally {
+            IOUtils.close(searcher);
+        }
+    }
+
+    @Override
+    protected void closeNoLock(String reason, CountDownLatch closedLatch) {
+        try {
+            IOUtils.close(lastCommitSearcher);
+        } catch (Exception ex) {
+            logger.warn("failed to close searcher", ex);
+        } finally {
+            closedLatch.countDown();
+        }
+    }
+
+    @Override
+    public GetResult get(Get get, BiFunction<String, SearcherScope, Searcher> searcherFactory) throws EngineException {
+        return getFromSearcher(get, this::acquireSearcher, SearcherScope.INTERNAL);
+    }
+
+    @Override
+    public Searcher acquireSearcher(String source, SearcherScope scope) throws EngineException {
+        store.incRef();
+        Releasable releasable = store::decRef;
+        try (ReleasableLock ignored = readLock.acquire()) {
+            final EngineSearcher searcher = new EngineSearcher(source, searcherManager, store, logger);
```

Comment:
can you try to exercise this method to make sure we open a new searcher and close / release everything

Reply:
Ah, `getDocIds` method in `SearchOnlyEngineTests#testSearchOnlyEngine` acquires searchers.